### PR TITLE
properly de/re-activate app bot accounts

### PR DESCRIPTION
**Pull request by levb**
https://mattermost.atlassian.net/browse/MM-34827

### CODE REVIEW

File: server/proxy/install.go [hunk 1]
```diff
@@ -152,61 +146,77 @@ func (p *Proxy) ensureBot(manifest *apps.Manifest, actingUserID string, client *
...
 		}
 	} else {
 		if !user.IsBot {
-			return nil, nil, errors.New("a user already owns the bot username")
+			return errors.New("a user already owns the bot username")
 		}
 
 		fullBot = model.BotFromUser(user)
-		if fullBot.DeleteAt != 0 {
+		if user.DeleteAt != 0 {
```
Comment (levb) [inline server/proxy/install.go:159]: !!!

**Review by hanzei — APPROVED** [review]
Thanks for fixing this annoying issue @levb :tada: The code LGTM

Two non blocking questions below.

**Review by mickmister — APPROVED** [review]
LGTM :+1:

**Review by DHaussermann — APPROVED** [review]
Tested and passed.
- Stated from a fresh make nuke 
- Repeated install / Uninstall cycle several times with Service Now 
- The same bot now activates and deactivates as expected
- No issues found that seem related 

LGTM!
Please merge.

**Pull request merged**